### PR TITLE
VBM-49: add root screen wrapper

### PR DESCRIPTION
**Pull request by natata7**

### CODE REVIEW

File: packages/mobile/src/components/screen-wrapper/screen-wrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React, { FC, ReactNode } from 'react';
+import { SafeAreaView } from '../components';
+import { styles } from './styles';
+
+type Props = {
+  children?: ReactNode;
+};
+
+const ScreenWrapper: FC<Props> = ({ children }) => {
+  return <SafeAreaView style={styles.container}>{children}</SafeAreaView>;
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/screen-wrapper/screen-wrapper.tsx:10]:
```suggestion
import { NativeSafeAreaViewProps } from 'react-native-safe-area-context';

type ScreenWrapperProps =  NativeSafeAreaViewProps;

const ScreenWrapper: FC<ScreenWrapperProps> = ({ style, children, ...restProps }) => {
  return <SafeAreaView style={[styles.container, style]} {...restProps}>{children}</SafeAreaView>;
```

File: packages/mobile/src/components/components.ts [hunk 2]
```diff
@@ -1,3 +1,3 @@
-export { Text, View } from 'react-native';
+export { Text, View, SafeAreaView } from 'react-native';
```
[the remark this answers is not in the document]
Comment (natata7) [reply packages/mobile/src/components/components.ts]: updated

File: packages/mobile/src/index.tsx [hunk 3]
```diff
@@ -2,15 +2,19 @@ import React, { FC } from 'react';
...
 import { store } from '~/store/store';
+import { styles } from './styles';
 
 const App: FC = () => {
   return (
     <StoreProvider store={store}>
-      <NavigationContainer>
-        <Navigation />
-      </NavigationContainer>
+      <SafeAreaView style={styles.rootContainer}>
```
[the remark this answers is not in the document]
Comment (natata7) [reply packages/mobile/src/index.tsx]: updated

File: packages/mobile/src/components/screen-wrapper/styles.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
```
[the remark this answers is not in the document]
Comment (natata7) [reply packages/mobile/src/components/screen-wrapper/styles.ts:6]: updated

File: packages/mobile/src/components/screen-wrapper/screen-wrapper.tsx (hunk 1 above)
Comment (natata7) [reply packages/mobile/src/components/screen-wrapper/screen-wrapper.tsx]: updated

File: packages/mobile/package.json [hunk 5]
```diff
@@ -32,7 +32,7 @@
     "react-native": "0.69.3",
     "react-native-config": "1.4.6",
     "react-native-mmkv": "2.4.3",
-    "react-native-safe-area-context": "4.3.1",
+    "react-native-safe-area-context": "^4.3.1",
```
Comment (AleksandrTermenzhy) [inline packages/mobile/package.json:35]: We don't use any modifiers.

```suggestion
    "react-native-safe-area-context": "4.3.1",
```

File: packages/mobile/src/components/screen-wrapper/screen-wrapper.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import { SafeAreaView } from '../components';
+import { styles } from './styles';
+
+const ScreenWrapper: FC<NativeSafeAreaViewProps> = ({
+  style,
+  children,
+  ...restProps
+}) => {
+  return (
+    <SafeAreaView style={styles.container} {...restProps}>
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/screen-wrapper/screen-wrapper.tsx:12]:
```suggestion
    <SafeAreaView style={[styles.container, style]} {...restProps}>
```

**Review by AleksandrTermenzhy — APPROVED** [review]

**Review by seal323 — APPROVED** [review]

**Pull request merged**